### ISSUE TITLE
`ImpossibleDeserialisationFailure` triggered

### ISSUE THREAD

**Issue by sourabhxyz** [IntersectMBO/plutus#5689]
### Summary

I am using `loadFromFile` for the attached UPLC program and seeing the following error:

```
Failed to deserialise our own program! This is a bug, please report it. Caused by: T
ooMuchSpace (0x0000000109d5abea,S {currPtr = 0x0000000109d56443, usedBits = 7})
```

UPLC program was generated using `-fplugin-opt PlutusTx.Plugin:dump-uplc` flag.

Please find attached the `contract.txt` file.
[contract.txt](https://github.com/IntersectMBO/plutus/files/13716518/contract.txt)

### Steps to reproduce the behavior

Already provided.

### Actual Result

Already provided.

### Expected Result

Should have correctly deserialised the serialised `CompiledCode`.

### System info

OS: MacOS
Version: Sonoma 14.1.2
Plutus library version: 1.6.0.0

**Comment by sourabhxyz** [comment]
Thanks for your reply @kwxm! I was able to obtain deserialised UPLC from the shared `contract.txt` file using the code mentioned [here](https://github.com/IntersectMBO/plutus/issues/5331#issuecomment-1567158003). However, when I tried to apply my argument, as mentioned [here](https://github.com/IntersectMBO/plutus/issues/5331#issuecomment-1864282136), I got a different script hash than what I had obtained if I had applied the argument to my initial `CompiledCode` (i.e., without serialising it and then deserialising it elsewhere).
